src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -247,12 +229,16 @@ void DeleteQueueIfPresent(string queueName)
         {
             var path = MsmqAddress.Parse(queueName).PathWithoutPrefix;
 
-            if (!MessageQueue.Exists(path))
+            if (MessageQueue.Exists(path))
             {
-                return;
+                MessageQueue.Delete(path);
             }
+        }
 
-            MessageQueue.Delete(path);
+        string GetReallyLongQueueName()
+        {
+            var maxQueueNameForSetPermissionToWork = 102 - Environment.MachineName.Length;
```

Comment:
@andreasohlund - I got to the bottom of it. 
TL;DR - Remove the unit test and get this PR merged.

>The thinking was that since we always call CreateQueueIfNecessary when installers are run (and we tell users to run them on every deploy of the endpoint) we need to make sure that it won't choke for existing queues since it would prevent existing users with to long queues to upgrade their endpoints.

>Does that logic make sense?

>If yes is the implementation correct based on the above?

Given our policy towards backwards compatibility, I'd agree. However here is some background on why we are not in danger of breaking backwards compatibility and why this test can be removed. 

1. According to the code, setting queue permissions can throw an exception when a long queue name is used, that exception is caught and logs a Warning and the overall method itself does not throw. 
3. The test attempts to cover this case. However, it creates a name such that the total format name is 123 characters. This will never fail the SetQueuePermissions method and therefore the unit test will never exercise the exception case where it just logs a warn.  To really test this method, the queue name should set to something that exceeds 124. In which case, the code will still create the queue, Queue permissions will fail, but the method will log a warn and not throw. The downside of that is we will now have a queue that cannot be deleted automatically. What we are currently testing is the safest longest length which is no different than testing with a queue named just foo. Both will succeed. 

Why we are not in danger of breaking backwards compatibility: 
Queues whose format name exceeds 123 will have issues such as trying to get permissions on the queue, getting properties on the queue etc. And this is because, in the Messaging .NET class, setting permissions invokes [MQPathNameToFormatName](https://msdn.microsoft.com/en-us/library/ms704126(v=vs.85).aspx) and the buffer size passed to it is 124. Therefore a long queue name can make the format name exceed the 124 buffer. The format name includes characters in addition to the queue name. An example:
`DIRECT=OS:[ComputerName]\\private$\\[QueueName]`. Therefore based on the Computer Name and the extra characters in the format name, this call can throw an exception. However, this is not just for QueuePermission, this call gets invoked for GetProperties, SetProperties as well. In short, you can't have a functional queue that's in this state in production already.  I also found this which supports what I found above:
http://geekswithblogs.net/Plumbersmate/archive/2011/01/26/msmq-error-ldquoformat-name-buffer-supplied-to-the-api-was.aspx

What I am advocating for this PR is to removing the unit test. The tear down will also become simpler getting rid of the 3rd queue. 
Whether we throw or when we warn, we can handle it in #4630.  

cc / @Particular/msmq-transport-maintainers @Particular/nservicebus-maintainers